app/Mage.php

```diff
@@ -751,6 +751,7 @@ public static function run($code = '', $type = 'store', $options = [])
             die();
         } catch (Exception $e) {
             if (self::isInstalled()) {
+                self::dispatchEvent('mage_installed_exception', ['exception' => $e]);
```

Comment:
```suggestion
            try {
                self::dispatchEvent('mage_run_exception', ['exception' => $e]);
                if (!headers_sent() && !self::isInstalled()) {
```

You can do a test easily with `throw new Exception('test');` in any controller action. Not sure how to test installation but the check on `!isInstalled()` seems correct, without the not, it would redirect wrongly.